### ISSUE TITLE
Bun.build output hash doesn't account for dynamic import

### ISSUE THREAD

**Issue by codehz** [oven-sh/bun#12428]
### What version of Bun is running?

1.1.18+5a0b93523

### What platform is your computer?

Darwin 23.5.0 arm64 arm

### What steps can reproduce the bug?

file build.ts:
```ts
const result = await Bun.build({
  entrypoints: ["1.ts", "2.ts"],
  splitting: true,
  minify: true
});

for (const output of result.outputs) {
  console.log(output.path, ":", output.hash);
  console.log(
    (await output.text())
      .split("\n")
      .map((x) => "| " + x)
      .join("\n")
  );
}
```
file 1.ts:
```ts
import("./3.ts");
```
file 2.ts:
```ts
import("./3.ts");
```
file 3.ts:
```ts
import { randomInt } from "crypto" with { type: 'macro' };

export const random = randomInt(100)
```

run build.ts with bun

### What is the expected behavior?

the generated hash for entrypoint file will change when 3.ts changed (it use random, so will change every time)

### What do you see instead?

got same hash for entrypoint file
```
./1.js : 8366f94ce3fac3cc
| import{a as i} from"./chunk-12ec8b15ea59dbbf.js";import("./chunk-647af94fa6f1c725.js");
| 
./2.js : 0a07085d284c9942
| import{a as i} from"./chunk-12ec8b15ea59dbbf.js";import("./chunk-647af94fa6f1c725.js");
| 
./chunk-647af94fa6f1c725.js : 647af94fa6f1c725
| import"./chunk-12ec8b15ea59dbbf.js";var o=61;export{o as random};
| 
./chunk-12ec8b15ea59dbbf.js : 12ec8b15ea59dbbf
| var d=((a)=>typeof require!=="undefined"?require:typeof Proxy!=="undefined"?new Proxy(a,{get:(b,c)=>(typeof require!=="undefined"?require:b)[c]}):a)(function(a){if(typeof require!=="undefined")return require.apply(this,arguments);throw Error('Dynamic require of "'+a+'" is not supported')});
| export{d as a};
| 
```

```
./1.js : 8366f94ce3fac3cc
| import{a as i} from"./chunk-12ec8b15ea59dbbf.js";import("./chunk-7e1613d5fcd5e8e0.js");
| 
./2.js : 0a07085d284c9942
| import{a as i} from"./chunk-12ec8b15ea59dbbf.js";import("./chunk-7e1613d5fcd5e8e0.js");
| 
./chunk-7e1613d5fcd5e8e0.js : 7e1613d5fcd5e8e0
| import"./chunk-12ec8b15ea59dbbf.js";var o=51;export{o as random};
| 
./chunk-12ec8b15ea59dbbf.js : 12ec8b15ea59dbbf
| var d=((a)=>typeof require!=="undefined"?require:typeof Proxy!=="undefined"?new Proxy(a,{get:(b,c)=>(typeof require!=="undefined"?require:b)[c]}):a)(function(a){if(typeof require!=="undefined")return require.apply(this,arguments);throw Error('Dynamic require of "'+a+'" is not supported')});
| export{d as a};
| 
```

### Additional information

I use randomInt is only for PoC, normal editing will also cause this bug

**Comment by Jarred-Sumner** [comment]
macros aren't yet part of the hash

**Comment by codehz** [comment]
new minimal example (it affect the default chunk file naming)

1.ts:
```ts
import("./2.ts");
```
2.ts:
```ts
import("./3.ts");
```
3.ts:
```ts
console.log(1) // change the number manually
```

use command: `bun build --outdir .build --splitting 1.ts`

you will get something like this:
```

  1.js                     0.10 KB

  3-c96b55aede2685f9.js    0.03 KB

  2-8219168fb473146a.js    0.10 KB

  1-10c8d92e4b9ccc65.js    0.38 KB

[3ms] bundle 3 modules
```

change the 3.ts and rebuild multiple times, you will get a bunch of 3-XXXX.js while 1/2.js keep same name
<img width="227" alt="image" src="https://github.com/oven-sh/bun/assets/13158903/e582eaf7-0e85-4dbb-a351-44210c7295eb">

**Comment by codehz** [comment]
This issue can cause the CDN to cache old chunk files (because the generated file names are the same).

**Comment by paulzakin** [comment]
I can replicate a similar(?) problem that @codehz mentioned. 

Problem #1: If you change a chunk that two entrypoints depend on, an odd thing occurs. The hash (in the path) for the chunk changes, as does the `import` path for it in each entrypoint. However, the hash (in the path) of the entrypoint does not change. This is problematic when you cache the entrypoint forever - the underlying chunk changes, but the entrypoint does not change as a response. Not sure if this is the intended behaviour? 

Problem #2: I get a similar problem in the chunks themselves. If you change the code in Chunk #3, which is imported in Chunk #2, which is imported into Chunk #1, here is what happens. The hash (in the path) for Chunk #3 is updated, as is the path in the `import` in Chunk #2. The hash (in the path) for Chunk #2 is updated, as is the path in the`import` in Chunk #1. However, the hash (in the path) for Chunk #1 is not updated.

The example for Problem #1:

```ts
const output = await Bun.build({
  entrypoints: [`entrypoint-1.ts`, `entrypoint-2.ts`],
  outdir: BUILD,
  target: 'browser',
  splitting: true,
  minify: false,
  naming: '[dir]/[name].[hash].[ext]',
})
```

```
// entry1.ts
import { bar } from './bar'
export const entry1 = () => {
  console.log('FOO')
  bar()
}
```

```ts
// entry2.ts
import { bar } from './bar'
export const entry2 = () => {
  console.log('FOO')
  bar()
}
```

```ts
// bar.ts
export const bar = () => {
  console.log('BAR')
  return 'BAR'
}
```

**Comment by codehz** [comment]
fixed in 1.1.21
